xarray/test/test_backends.py

```diff
@@ -1063,3 +1063,131 @@ def test_extract_h5nc_encoding(self):
                           {'least_sigificant_digit': 2})
         with self.assertRaisesRegexp(ValueError, 'unexpected encoding'):
             _extract_nc4_encoding(var, raise_on_invalid=True)
+
+
+class MiscObject:
+    pass
+
+
+class TestValidateAttrs(TestCase):
+
+    def test_invalid_attr_names(self):
+        ds = Dataset({'data': ('y', np.arange(10.0))})
+        self.assert_invalid_attr_names_for_obj(ds.attrs, ds)
```

Comment:
I would suggest moving this helper function inside this method -- that makes it easier to follow the logic of these tests.